android/java/org/chromium/chrome/browser/settings/BraveNewsPreferences.java

```diff
@@ -74,20 +89,69 @@ public void onCreatePreferences(@Nullable Bundle savedInstanceState, String root
         mBraveNewsController.getPublishers((publishers) -> {
             List<Publisher> allPublishers = new ArrayList<>();
             List<Publisher> categoryPublishers = new ArrayList<>();
+            List<Publisher> rssPublishers = new ArrayList<>();
             for (Map.Entry<String, Publisher> entry : publishers.entrySet()) {
                 Publisher publisher = entry.getValue();
-                categoryPublishers.add(publisher);
-                mCategsPublishers.put(publisher.categoryName, categoryPublishers);
+                if (publisher.type != PublisherType.DIRECT_SOURCE) {
+                    categoryPublishers.add(publisher);
+                    mCategsPublishers.put(publisher.categoryName, categoryPublishers);
+                } else {
+                    rssPublishers.add(publisher);
+                }
             }
             mCategsPublishers.put("All Sources", allPublishers);
             addCategs(mCategsPublishers);
+            addRss(rssPublishers);
         });
     }
 
+    private void addRss(List<Publisher> publishers) {
+        for (Publisher publisher : publishers) {
+            if (publisher.type != PublisherType.DIRECT_SOURCE) {
+                return;
```

Comment:
as per https://github.com/brave/brave-core/pull/12379/files#diff-2ddfa5e09ee57aab48d02c29d3a82bc7f71f4c1e3eeb94559e83e06c047a4d45R99
`publishers` contains only those which are `publisher.type == PublisherType.DIRECT_SOURCE`
so this `return` statement will not be executed ever.

What is the intention of this check?
Maybe it is enough to put `assert(publisher.type == PublisherType.DIRECT_SOURCE)`?

Reply:
@AlexeyBarabash yeah that was exactly the point. a double check that only those sources are added to the RSS section